terraform/main.tf

```diff
@@ -0,0 +1,96 @@
+# INSTRUCTIONS: 
+# 1) ENSURE YOU POPULATE THE LOCALS 
+# 2) ENSURE YOU REPLACE ALL INPUT PARAMETERS, THAT CURRENTLY STATE 'ENTER VALUE', WITH VALID VALUES 
+# 3) YOUR CODE WOULD NOT COMPILE IF STEP NUMBER 2 IS NOT PERFORMED!
+# 4) ENSURE YOU CREATE A BUCKET FOR YOUR STATE FILE AND YOU ADD THE NAME BELOW - MAINTAINING THE STATE OF THE INFRASTRUCTURE YOU CREATE IS ESSENTIAL - FOR APIS, THE BUCKETS ALREADY EXIST
+# 5) THE VALUES OF THE COMMON COMPONENTS THAT YOU WILL NEED ARE PROVIDED IN THE COMMENTS
+# 6) IF ADDITIONAL RESOURCES ARE REQUIRED BY YOUR API, ADD THEM TO THIS FILE
+# 7) ENSURE THIS FILE IS PLACED WITHIN A 'terraform' FOLDER LOCATED AT THE ROOT PROJECT DIRECTORY
```

Comment:
We could probably remove these comments now the instructions have been followed

Reply:
I personally feel that we should leave them for now as it is just the common instructions and also, I did the changes now but in the future, when someone who is not familiar with the set up make changes to this file, it might be useful for them to understand what needs to be done/why has certain things been done.